nixops/resources/output.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+
+# Arbitrary JSON.
+
+import os
+import nixops.util
+import nixops.resources
+
+import tempfile
+import shutil
+import subprocess
+import hashlib
+from typing import Any, Optional, List, Dict, AnyStr, Tuple
+
+class OutputDefinition(nixops.resources.ResourceDefinition):
+    """Definition of an Output."""
+
+    @classmethod
+    def get_type(cls):
+        # type: () -> (str)
+        return "output"
+
+    @classmethod
+    def get_resource_type(cls):
+        # type: () -> (str)
+        return "output"
+
+    def __init__(self, xml):
+        # type: (Any) -> None
+        nixops.resources.ResourceDefinition.__init__(self, xml)
+        self.script = xml.find("attrs/attr[@name='script']/string").get("value")
+        self.name = xml.find("attrs/attr[@name='name']/string").get("value")
+
+    def show_type(self):
+        # type: () -> (str)
+        return "{0}".format(self.get_type())
+
+class OutputState(nixops.resources.ResourceState):
+    """State of an output."""
+    state = nixops.util.attr_property("state", nixops.resources.ResourceState.MISSING, int)
+    script = nixops.util.attr_property("script",None)
+    value = nixops.util.attr_property("value",None)
+    name =  nixops.util.attr_property("name",None)
+
+    @classmethod
+    def get_type(cls):
+        # type: () -> (str)
+        return "output"
+
+    @property
+    def resource_id(self):
+        # type: () -> Optional[str]
+        if self.value is not None:
+            # Avoid printing any potential secret information
+            return "{0}-{1}".format(self.name,hashlib.sha256(self.value).hexdigest()[:32])
+        else:
+            return None
+
+    def __init__(self, depl, name, id):
+        # type: (Any,str,int) -> None
+        # Any is: nixops.deployment.Deployment
```

Comment:
You can actually use import for `nixops.deployment.Deployment` instead of `Any`.

Reply:
fixed